pylint/checkers/classes/class_checker.py

```diff
@@ -192,7 +195,9 @@ def _is_trivial_super_delegation(function: nodes.FunctionDef) -> bool:
 # Deal with parameters overriding in two methods.
 
 
-def _positional_parameters(method):
+def _positional_parameters(
+    method: bases.BoundMethod | bases.UnboundMethod | nodes.FunctionDef,
+) -> list[nodes.AssignName]:
```

Comment:
This can also be `None`. See the docstring of `args.args` in `astroid`.

Reply:
There's a guard against that earlier

https://github.com/PyCQA/pylint/blob/6417e8cdee987afb773ad5ab9f5f0e82028c7cac/pylint/checkers/classes/class_checker.py#L2096

Then we call ``_different_parameters`` :

https://github.com/PyCQA/pylint/blob/6417e8cdee987afb773ad5ab9f5f0e82028c7cac/pylint/checkers/classes/class_checker.py#L2107

Then ``_positional_parameters``

https://github.com/PyCQA/pylint/blob/6417e8cdee987afb773ad5ab9f5f0e82028c7cac/pylint/checkers/classes/class_checker.py#L321

https://github.com/PyCQA/pylint/blob/6417e8cdee987afb773ad5ab9f5f0e82028c7cac/pylint/checkers/classes/class_checker.py#L322

So effectively it can only be a ``list[nodes.AssignName]``. Maybe we need to add an assert there to prevent the function being used in another context without the check ?

Reply:
Here's what mypy raise if we add the ``None``:
```
pylint/checkers/classes/class_checker.py:338: error: Item "None" of "Optional[List[Any]]" has no attribute "__iter__" (not iterable)  [union-attr]
pylint/checkers/classes/class_checker.py:340: error: Item "None" of "Optional[List[Any]]" has no attribute "__iter__" (not iterable)  [union-attr]
pylint/checkers/classes/class_checker.py:350: error: Argument 1 to "_has_different_parameters" has incompatible type "Optional[List[Any]]"; expected "List[Any]"  [arg-type]
pylint/checkers/classes/class_checker.py:350: error: Argument 2 to "_has_different_parameters" has incompatible type "Optional[List[Any]]"; expected "List[Any]"  [arg-type]
```

Reply:
Ah yeah. Sorry, should have seen that. Then this typing is very useful 😄

Reply:
Yeah, that's a 10mn expedition using a good IDE I don't think you're supposed to just 'see' it :smile: